woocommerce/woocommerce
# Update/34885 category field in product editor

### All Submissions:






### Changes proposed in this Pull Request:

This updates the product category metabox with the current product editor with a react version of it that replaced the list of categories with a async search dropdown.
The async search dropdown makes use of the Tree Select Control component.

Closes #34885  .

https://user-images.githubusercontent.com/2240960/220586001-39acc0f1-8adf-4170-9050-0447746548db.mp4






### How to test the changes in this Pull Request:




1. Enable the `async-product-editor-category-field` feature flag using the WooCommerce Beta Tester.
2. Load this branch and create a new store and be sure to create some new Categories.
3. Go to **Products > Add New**
4. The usual category metabox should be present if you have less then a 100 categories.
5. Create either more then a 100 categories or add this filter to your `mu-plugin` -> `add_filter( 'woocommerce_product_category_metabox_search_threshold', function() { return 5; } );`
6. Add a name and scroll down to the Product Categories metabox on the left hand side
7. Try search for some of the categories you previously added and select these.
8. Save the product, the categories should persist.
9. Other items to check:
     - Selecting a popular category should automatically select the category within the popular tab and vice versa
     - Creating a new category should automatically add the new item to the selected list.
     - Creating a new category and selecting a parent should work correctly.



### Other information:

-   [] Have you added an explanation of what your changes do and why you'd like us to include them?



### FOR PR REVIEWER ONLY:
